### PR TITLE
#546 Basic stocktake detail view

### LINKED ISSUE #546: Basic Stocktake detail page
Just want to add the boiler plate/basic stock take detail page code to get it out of the way

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #546 

Just some more boilerplate for a basic stocktake detail view

### CODE REVIEW

**Comment by joshxg** [comment]
That is very cool :)

File: packages/inventory/src/types.ts [hunk 1]
```diff
@@ -9,3 +10,35 @@ export type StocktakeRow = Pick<
...
+}
+
+export interface Stocktake extends Omit<StocktakeNode, 'lines' | '__typename'> {
+  lines: StocktakeLine[];
+}
+
+export interface StocktakeController extends Omit<Stocktake, 'lines'> {
+  lines: StocktakeItem[];
+  update: (key: string, value: string) => void;
+}
```
Comment (joshxg) [inline packages/inventory/src/types.ts:35]: Hmm.

Wasn't quite sure here. Normally we have `RequisitionNode` -> `Requisition` -> `CustomerRequisition` and the same for invoices. But here it's `StocktakeNode` -> `Stocktake` -> `StocktakeController` which isn't great naming

**Review by joshxg — COMMENTED** [review]
It's all pretty standard copy paste nothing much to add to try help sorry!

File: packages/inventory/src/types.ts (hunk 1 above)
Comment (mark-prins) [reply packages/inventory/src/types.ts:35]: I've carefully considered and have managed to convince myself that the naming is fine 😁 
Basically, I have no better suggestions

File: packages/inventory/src/utils.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  );
+
+  const nextStatus = statuses[currentStatusIdx + 1];
+
+  if (!nextStatus) throw new Error('Could not find the next status');
+
+  return nextStatus;
+};
+
+// TODO: When supplier requisition statuses are finalised, this function should be passed
```
Comment (mark-prins) [inline packages/inventory/src/utils.ts:50]: interesting that this one has `requisition` and the other comments were invoice related. Anyway, it's a minor observation only, doesn't affect anything

**Review by mark-prins — APPROVED** [review]
😁  great! I haven't contributed much that is helpful, just nit picking over naming.

File: packages/inventory/src/Stocktake/DetailView/SidePanel.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      title={t('heading.related-documents', { ns: 'distribution' })}
+    >
+      <Grid container gap={0.5} key="additional-info"></Grid>
+    </DetailPanelSection>
+  );
+};
+
+export const SidePanel: FC<SidePanelProps> = ({ draft }) => {
+  const { success } = useNotification();
+  const t = useTranslation(['outbound-shipment', 'common']);
```
[the remark this answers is not in the document]
Comment (joshxg) [reply packages/inventory/src/Stocktake/DetailView/SidePanel.tsx:51]: Yup - I've made an issue to get it all done heh

**Pull request merged**